compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -3276,23 +3277,29 @@ private void resolveFields(BStructureType structureType, BLangStructureTypeNode
                 })
                 .collect(getFieldCollector());
 
-        List<BType> list = new ArrayList<>();
+        // Resolve referenced types and their fields of structural type
+        resolveReferencedFields(structureTypeNode, typeDefEnv);
```

Comment:
Any particular reason to change the order here?

Reply:
Yes because they are not resolved yet. We are collecting invalid types. Then we try to write them 
```writeTypeInclusions(bRecordType.typeInclusions);```
If you look at the same logic in BLangClassDefinition visitor in SymbolEnter we only collect valid refs.